### PR TITLE
Collage photo module.

### PR DESCRIPTION
**Pull request by jerdzica**

### CODE REVIEW

File: addons/photo_collage/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+{
+    'name': "Photo Collage",
+    'summary': """Controller for the photo collage""",
+    'description': """
+        Controller for the photo collage
```
Comment (ludwiktrammer) [inline addons/photo_collage/__openerp__.py]: It's a full module with its own forms, views etc., so I wouldn't describe it as just a "controller".

File: addons/photo_collage/controllers.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from openerp import http
+
+
+class PhotoCollage(http.Controller):
+
+    @http.route('/photo_collage/objects/', auth='public', website=True)
```
Comment (ludwiktrammer) [inline addons/photo_collage/controllers.py]: "/objects" seems redundant here.

File: addons/photo_collage/controllers.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    def list(self, **kw):
```
Comment (ludwiktrammer) [inline addons/photo_collage/controllers.py]: There is a very strong convention to always call this "kwargs".

File: addons/photo_collage/models.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+class PhotoCollage(models.Model):
+    _name = 'photo_collage'
+
+    name = fields.Char(string="Nazwa", required=True)
+    reverse = fields.Text(string="Tekst na rewersie")
+    attachment = fields.Binary(required=True, string="Plik")
+    filename = fields.Char()
+
+    @api.multi
+    def get_public_url(self):
```
Comment (ludwiktrammer) [inline addons/photo_collage/models.py]: This doesn't seem relevant now.

File: addons/photo_collage/models.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        return '/photo_collage/{}/'.format(slug(self))
+
+    @api.multi
+    def show_website_action(self):
```
Comment (ludwiktrammer) [inline addons/photo_collage/models.py]: This doesn't seem relevant now too ;)

File: addons/photo_collage/security/ir.model.access.csv [hunk 6]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_photo_collage,photo_collage.photo_collage,model_photo_collage,base.group_user,1,0,0,0
```
Comment (ludwiktrammer) [inline addons/photo_collage/security/ir.model.access.csv]: `base.group_user` user would prevent logged out users from viewing those objects. Leave the "group_id" field empty.

File: addons/photo_collage/templates.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<openerp>
+    <data>
+        <template id="listing">
+           <t t-call="website.layout">
+              <ul>
+                <li t-foreach="objects" t-as="photo_collage">
+                    <div class="row">
+                        <img t-attf-src="/website/image/photo_collage/{{photo_collage.id}}/attachment"/>
```
Comment (ludwiktrammer) [inline addons/photo_collage/templates.xml]: Where did you get the `/attachment` part from? I don't see this as an option for the `/website/image/` controller...

Comment (jerdzica) [inline addons/photo_collage/templates.xml]: This is a field in my model, "attachment" is the actual picture.

File: addons/photo_collage/views/photo_collage.xml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                            <label for="name" class="oe_edit_only"/>
+                            <h1 class="oe_inline">
+                                <field name="name"/>
+                            </h1>
+                        </div>
+                        <group col="2">
+                            <label for="attachment"/>
+                            <div>
+                                <field name="attachment" filename="filename" class="no_clear_button"/>
+                                <div class="oe_grey oe_text_right oe_edit_only">Max. wielkość pliku do <strong>25 MB</strong></div>
```
Comment (KrzysiekJ) [inline addons/photo_collage/views/photo_collage.xml]: Does the value “25 MB” need to be hardcoded – isn’t it an Odoo preference that can be extracted?

Comment (ludwiktrammer) [inline addons/photo_collage/views/photo_collage.xml]: From what I remember I'm pretty sure it is hardcoded somewhere in Odoo.

Comment (jerdzica) [inline addons/photo_collage/views/photo_collage.xml]: I took it from bestja_files, there is such a limit. If you would like to have it changed, you may need to change it there as well.

**Pull request merged**